### PR TITLE
adding license at the end of every file for CC-BY-SA

### PR DESCRIPTION
**Pull request by leios**
I, James Schloss, the copyright holder of this work, irrevocably grant anyone the right to use the Algorithm Archive, including derivatives created during the review process, under the Creative Commons Attribution ShareAlike 4.0 license ([legal code](https://creativecommons.org/licenses/by-sa/4.0/deed.en)). (Anyone may use, share or remix this work, as long as they credit the appropriate author and share any derivative work under this license.)

This means that all previous versions of the Algorithm Archive before this point are still under CC-NC, but all versions beyond this point will be appropriately attributed to the author of each chapter and chapter editors.

Full discussion can be found in the #493.

### CODE REVIEW

**Comment by julianschacher** [comment]
I think we need to modify the license section a bit to also include graphics and PRs and be more structured. I thought of something like this for the License section:

## License

### Text
The text of this of this chapter by PERSON, PERSON, ... and PERSON is licensed under [CC BY-SA 4.0](https://creativecommons.org/licenses/by-sa/4.0/legalcode).

### Code
The code examples are licensed under the MIT license (found in [LICENSE.md](https://github.com/algorithm-archivists/algorithm-archive/blob/master/LICENSE.md)).

### Images/Graphics
- The first image "IMAGENAME" by PERSON is licensed under [CC BY-SA 4.0](https://creativecommons.org/licenses/by-sa/4.0/legalcode).
- The second image "IMAGENAME" by PERSON is licensed under [CC BY-SA 4.0](https://creativecommons.org/licenses/by-sa/4.0/legalcode).
...
- The nth image "IMAGENAME" by PERSON is licensed under [CC BY-SA 4.0](https://creativecommons.org/licenses/by-sa/4.0/legalcode).

### Pull Requests
The following Pull Requests changed this chapter:
- [#560](https://github.com/algorithm-archivists/algorithm-archive/pull/560)
- PR
...

---

### Changes

- I changed the wording for the attribution a bit. This way it's shorter and complies with <https://wiki.creativecommons.org/wiki/Best_practices_for_attribution>
- The new images/graphics section allows us to credit the author of each image/graphic properly.
- The new PR section enables the reader to easily comprehend the progress of the chapter creation.
- The general structure allows for an easier overview.

To comply with <https://wiki.creativecommons.org/wiki/Best_practices_for_attribution> (and since it's good to do so) we should also try to make IMAGENAME and PERSON links, if possible.
For new chapter contributions we can then ask each person easily how they want to be linked.

**Comment by julianschacher** [comment]
Since two of the images were created by me, more specifically: "split_op_method" and "huffman_tree", here's also my statement (I don't know, if it's really needed, but it doesn't hurt for sure).

I, Julian Schacher, the copyright holder of the images "[split_op_method](https://github.com/algorithm-archivists/algorithm-archive/blob/master/contents/split-operator_method/res/split_op_method.svg)" and "[huffman_tree](https://github.com/algorithm-archivists/algorithm-archive/blob/master/contents/huffman_encoding/res/huffman_tree.png)", irrevocably grant anyone the right to use this work, including derivatives created during the review process, under the Creative Commons Attribution ShareAlike 4.0 license ([legal code](https://creativecommons.org/licenses/by-sa/4.0/deed.en)).

(Anyone may use, share or remix this work, as long as they credit me and share any derivative work under this license.)

File: contents/euclidean_algorithm/euclidean_algorithm.md [hunk 1]
```diff
@@ -238,3 +238,25 @@ A text version of the program is provided for both versions.
...
+
+##### Text
+
+The text of this chapter was written by [James Schloss](https://github.com/leios) and is licensed under the [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/legalcode).
+
+[<p><img  class="center" src="../cc/CC-BY-SA_icon.svg" /></p>](https://creativecommons.org/licenses/by-sa/4.0/)
+
+##### Images/Graphics
+- The image "[Euclidsub](res/subtraction.png)" was created by [James Schloss](https://github.com/leios) and is licenced under the [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/legalcode).
+- The image "[Euclidmod](res/subtraction.png)" was created by [James Schloss](https://github.com/leios) and is licenced under the [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/legalcode).
```
Comment (julianschacher) [inline contents/euclidean_algorithm/euclidean_algorithm.md]: "Euclidmod" needs to link to `modulus.png`.
```suggestion
- The image "[Euclidmod](res/modulus.png)" was created by [James Schloss](https://github.com/leios) and is licenced under the [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/legalcode).
```

File: contents/cooley_tukey/cooley_tukey.md [hunk 2]
```diff
@@ -242,3 +242,27 @@ Some rather impressive scratch code was submitted by Jie and can be found here:
...
+
+[<p><img  class="center" src="../cc/CC-BY-SA_icon.svg" /></p>](https://creativecommons.org/licenses/by-sa/4.0/)
+
+##### Images/Graphics
+- The image "[FTexample](res/FT_example.png)" was created by [James Schloss](https://github.com/leios) and is licenced under the [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/legalcode).
+- The image "[radix2positive](res/radix-2screen_positive.jpg)" was created by [James Schloss](https://github.com/leios) and is licenced under the [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/legalcode).
+- The image "[radix2](res/radix-2screen.jpg)" was created by [James Schloss](https://github.com/leios) and is licenced under the [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/legalcode).
+- The image "[radix8](res/radix-8_screen.jpg)" was created by [James Schloss](https://github.com/leios) and is licenced under the [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/legalcode).
```
Comment (julianschacher) [inline contents/cooley_tukey/cooley_tukey.md]: Link doesn't work due to typo.
```suggestion
- The image "[radix8](res/radix-8screen.jpg)" was created by [James Schloss](https://github.com/leios) and is licenced under the [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/legalcode).
```

**Comment by leios** [comment]
Cool, thanks for catching those!

**Pull request merged**